modules/eks/main.tf

```diff
@@ -45,3 +65,197 @@ module "eks" {
+  count = var.enable_nvme_storage ? 1 : 0
+
+  metadata {
+    name      = "nvme-disk-setup"
+    namespace = "kube-system"
+  }
+
+  spec {
+    selector {
+      match_labels = {
+        app = "nvme-disk-setup"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "nvme-disk-setup"
+        }
+      }
+
+      spec {
+        service_account_name = kubernetes_service_account.nvme_setup[0].metadata[0].name
+
+        # Use init container to set up NVMe disks
+        init_container {
+          name  = "setup-nvme"
+          image = var.nvme_bootstrap_image
+
+          security_context {
+            privileged = true
+          }
+
+          env {
+            name = "NODE_NAME"
+            value_from {
+              field_ref {
+                field_path = "spec.nodeName"
+              }
+            }
+          }
+
+          env {
+            name  = "CLOUD_PROVIDER"
+            value = "aws"
+          }
+
+          volume_mount {
+            name       = "host-dev"
+            mount_path = "/dev"
+          }
+
+          volume_mount {
+            name       = "host-sys"
+            mount_path = "/sys"
+          }
+
+          volume_mount {
+            name       = "host-proc"
+            mount_path = "/proc"
+          }
+
+          command = [
+            "/bin/bash",
+            "-c",
+            <<-EOT
+            # Run the disk configuration script with explicit cloud provider
+            /usr/local/bin/configure-disks.sh --cloud-provider aws
+
+            # Remove taint once configuration is complete
+            /usr/local/bin/manage-taints.sh remove
+            EOT
+          ]
+        }
+
+        # Main container is just a pause container
+        container {
+          name  = "pause"
+          image = "k8s.gcr.io/pause:3.7"
+        }
+
+        volume {
+          name = "host-dev"
+          host_path {
+            path = "/dev"
+          }
+        }
+
+        volume {
+          name = "host-sys"
+          host_path {
+            path = "/sys"
+          }
+        }
+
+        volume {
+          name = "host-proc"
+          host_path {
+            path = "/proc"
+          }
+        }
+
+        # Allow this DaemonSet to run on nodes with the disk-unconfigured taint
+        toleration {
+          key      = "disk-unconfigured"
+          operator = "Exists"
+          effect   = "NoSchedule"
+        }
+
+        # Only run on nodes that need disk configuration
+        node_selector = {
+          "materialize.cloud/disk-config-required" = "true"
+        }
+      }
+    }
+  }
+
+  depends_on = [
+    module.eks,
+    kubernetes_cluster_role_binding.nvme_setup_taint_binding
+  ]
+}
+
+# Install OpenEBS for lgalloc support
+resource "kubernetes_namespace" "openebs" {
+  count = var.install_openebs ? 1 : 0
+
+  metadata {
+    name = var.openebs_namespace
+  }
+}
+
+resource "helm_release" "openebs" {
```

Comment:
Open to suggestions if we should have this in the EKS module or create a new separate submodules to handle any other Kubernetes installs.